### ISSUE TITLE
Error notice on psaddonsconnect - BO

### ISSUE THREAD

**Issue by Marwazeinoo88** [PrestaShop/PrestaShop#10934]
**Hi,  I have php issues display when i connect to my backoffice. Here are the error messages:**

Notice à la ligne 227 du fichier /home/www/modules/psaddonsconnect/psaddonsconnect.php
[8] Undefined variable: advice

Notice à la ligne 228 du fichier /home/www/modules/psaddonsconnect/psaddonsconnect.php
[8] Undefined variable: link_advice

Can you help to fix that please?
**This is the lines on the file:**
Line227:  'advice' => $advice,
Line228:  'link_advice' => $link_advice,

**Additionnal information**
PrestaShop version: 1.7.3.0
PHP version:  7.1.21
Thème actif : classic

![github](https://user-images.githubusercontent.com/43037924/46662766-99d58e80-cbbc-11e8-8d09-8b5ee722a791.PNG)
![github2](https://user-images.githubusercontent.com/43037924/46662915-f769db00-cbbc-11e8-9a98-febbb37296bd.PNG)

Thank you for your reply and help

**Comment by khouloudbelguith** [comment]
Hi @Marwazeinoo88,

I manage to reproduce this issue.
When I access the BO, the red alert is displayed.
This issue is randomly.
![image](https://user-images.githubusercontent.com/16067358/52471584-87cb2300-2b91-11e9-9580-2c53183303c0.png)
Reported here: http://forge.prestashop.com/browse/BOOM-5781
Thanks!

**Comment by Marwazeinoo88** [comment]
So for now,  i should just rename the folder "psaddonsconnect" (in www/modules/)?

**Comment by khouloudbelguith** [comment]
@Marwazeinoo88, you can navigate to your BO => IMPROVE => Modules => Modules & Services => search for this module & disable it.
Thanks!

**Comment by Marwazeinoo88** [comment]
okey, Thank you for your reply and help

**Comment by masiorama** [comment]
I just had the same problem with 1.7.4.

**Comment by khouloudbelguith** [comment]
Hi @masiorama,

Have you tried to disable the psaddonsconnect module?
PS: this issue is added to our bug roadmap, it is not fixed yet.
Thanks!

**Comment by masiorama** [comment]
I did @khouloudbelguith , and that specific error on login disappeared. Still today I'm experiencing so many difficulties with prestashop, expecially in getting modules updates... does it mean anything to you?
Thanks in advance.

**Comment by khouloudbelguith** [comment]
@masiorama, This issue is a different issue, can you open a new one with more information about your configuration & your different issue.
Thanks!

**Comment by masiorama** [comment]
Ok thanks I'll try to gather infos and open an issue... I'm kinda lost anyway, it seems everything random, I'll do the best I can.

**Comment by masiorama** [comment]
@khouloudbelguith just a brief update: during the last 3 days the problem was simply gone, so I don't see a reason to investigate more in depth. I'm pretty sure it was because of some problem with PS sending infos about modules updates, plus maybe my server behavior (which was pretty slow). We managed to upgrade to a different server and now everything seems solid. I will definitely open an issue in case I'll find mysql involved in this problem again. Thanks.

**Comment by jomdacillo** [comment]
Hello Guys,

PS version: 1.6.1.23

Just encountered this issue today. I believed it is a Prestashop bug, the error is a connection from Prestashop server. It prompt's an error in the dashboard because some of the variables has no values the $advice and the $link_advice.

So this is my quick fix for the issue while Prestashop haven't provide any fix for it.
1. Go to your root website folder
2. Navigate to /modules/psaddonsconnect/
3. Look for psaddonsconnect.php
4. Add an "else" statement so that we can still fill value for the variables $advice and $link_advice. (See attached code image)
![Screen Shot 2019-05-15 at 8 57 59](https://user-images.githubusercontent.com/33541221/57740101-0f748080-76f1-11e9-9b84-81659f7f4dc8.png)

Hope this helps.

Cheers!

**Comment by fraria76** [comment]
Hi everyone,

I have the same problem too.

I have this configuration:

- Windows Server 2019 with IIS 10
- Prestashop 1.7.5.1
- MySql 5.6.16.0
- PhpMyAdmin 4.8.5
- PHP 7.2.18

I have always connected to the BO side without any difficulty but since yesterday I do not access any more.

I can enter user and password but then a page appears with:

PHP Notice:  Undefined variable: advice in C:\inetpub\wwwroot\modules\psaddonsconnect\psaddonsconnect.php on line 227
PHP Notice:  Undefined variable: link_advice in C:\inetpub\wwwroot\modules\psaddonsconnect\psaddonsconnect.php on line 228

If I insert the code suggested by jomdacillo now I can log in.

But now if I want disable the psaddonsconnect module, what can I search on PrestaShop Addons Marketplace? (if i look for psaddonsconnect it finds nothing)

Kind regards

**Comment by evox95** [comment]
Create file /override/modules/psaddonsconnect/psaddonsconnect.php with content:
```php
<?php

class PsaddonsconnectOverride extends Psaddonsconnect
{

    /**
     * @param array $params
     * @return string
     * @throws SmartyException
     */
    public function hookDashboardZoneOne($params)
    {
        $this->loadAsset();

        $url = "";
        $logged_on_addons17 = 0;

        if ((bool)version_compare(_PS_VERSION_, '1.7', '>=')) {
            global $kernel;

            //router to get the addons URL
            $instance = $kernel->getContainer();
            $router = $instance->get('router');
            $url = $router->generate('admin_addons_login');

            if (isset($_COOKIE['username_addons'])) {
                $logged_on_addons17 = 1;
            }
        }

        // API DATA
        include_once(_PS_MODULE_DIR_ . 'psaddonsconnect/classes/WeekAdvice.php');
        $api = new WeekAdvice();
        $api_json = $api->getWeekData();

        if (!empty($api_json)) {
            $advice = $api_json->advice;
            $link_advice = $api_json->link;
        }

        // assign var to smarty
        $this->context->smarty->assign(array(
            'img_path' => $this->img_path,
            'ps_version' => (bool)version_compare(_PS_VERSION_, '1.7', '>='),
            'advice' => isset($advice) ? $advice : null,
            'link_advice' => isset($link_advice) ? $link_advice : null,
            'url_connexion' => $url,
            'logged_on_addons17' => $logged_on_addons17,
            'practical_links' => $this->practicalLinks(),
        ));

        $this->output .= $this->context->smarty->fetch(
            $this->local_path . 'views/templates/hook/dashboard_zone_one.tpl'
        );

        return $this->output;
    }

}
```
and remove files:
`/var/cache/dev/class_index.php`
`/var/cache/prod/class_index.php`

**Comment by andymorales** [comment]
Hi,

I had to disable the module for the bad way: changing directory name.

But no more error notice on sign in

Hope I helped.

**Comment by hitinder** [comment]
I am facing the same issue on 1.7.5.1 version of Prestashop. I am guessing I got this issue after I clicked on the google link showed in below image.

![image](https://user-images.githubusercontent.com/9841229/57987910-6501b200-7a90-11e9-9189-89234f2ddaff.png)

I am now not able to go to **Module Catalog** and when I click on **Module Manager** it gives me **internal error 500**

Hope this helps someone.

**Comment by hitinder** [comment]
Hey Guys! I enabled debug mode from the BO and tried accessing Module Manager again and this time I get exception details related to another plugin (https://github.com/cashfree/).

Here are few things that are important.
1. I copied (not installed) CashFree almost 3 weeks back and I was still able to access Module Manager
2. About 3 days back I clicked on the "Link to your google analytics account" and since then I started getting this error.
3. I removed the CashFree folder from Modules and I am now able to access Module Manager and install new modules.

P.S. None of the above solution worked for me, probably because the issue was from CashFree.

**Comment by Quetzacoalt91** [comment]
The notices have been fixed with the v2.0.0 of the module which was released today. Thanks for the report.

**Comment by arafatx** [comment]
Sorry for disturbing a closed issue. This still happens in prestashop 1.7.6.

**Comment by khouloudbelguith** [comment]
Hi @arafatx,

This issue is fixed in the `psaddonsconnect `module v2.0.0.
In my case it is ok.


Thanks!

**Comment by arafatx** [comment]
@khouloudbelguith I'm using v2.0.0. Strange behaviour is, when I put prestashop into sub domain with the same setup it works:

www.mywebsite.com - got error
dev.mywebsite.com - no error

Same setup files clone from github. Maybe this happen with specific domain ?

**Comment by arafatx** [comment]
Hi, sorry for making everybody panic. It was the cache issue! Problem solved! Thanks a lot.

**Comment by khouloudbelguith** [comment]
@arafatx, Try to access to this link: www.mywebsite.com/modules/psaddonsconnect/config.xml & check the version of this module.

Thanks!

**Comment by khouloudbelguith** [comment]
@arafatx, great!
Thanks!

**Comment by arafatx** [comment]
Sorry for commenting on the closed issue but I don't think this issue is fixed and related to cache. I got the same error again on the latest 1.7.6.1 (production). This happen randomly and I guess @jomdacillo is right this issue perhaps related to prestashop server. If i rename or remove the psaddonsconnect there is no problem connecting to Back Office.

**Comment by khouloudbelguith** [comment]
Hi @arafatx,

The notices have been fixed with the v2.0.0 of the module psaddonsconnect.
The last release of this module is v2.1.0.

Thanks!

**Comment by arafatx** [comment]
I guess this is random. I just went away from my PC and coming back (refresh the back office page) and it works. Didn't do anything.

**Comment by khouloudbelguith** [comment]
@arafatx, since the upgrade of the module `psaddonsconnect`, this issue doesn't occur in my case.
Could you please navigate to your BO => Module Manager page & search for the `psaddonsconnect` & check the exact version.
![image](https://user-images.githubusercontent.com/16067358/65678934-1a023780-e04c-11e9-975a-03d585311d01.png)

Thanks!

**Comment by arafatx** [comment]
The exact same version 2.1.0. I thought the error message was the same (sorry for this. This might be different issue related to CacheStorage):

I only got this error when navigating to BO -> Dashboard.

![cache-storage](https://user-images.githubusercontent.com/4420157/65680238-26f14c00-e089-11e9-9693-12148cfbb4d9.JPG)


Perhaps, I should open a new issue? Sorry and TQ.

- I tried clear the cache from BO - not working
- removed the prod and dev cache folder - not working

PS 1.7.6.1
